pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -85,10 +85,19 @@ func CreateEmbeddedTarget(dep rv1beta2.DeploymentInterface, sa rv1.ServiceAccoun
 				dep.Delete(loggingconfig.EmbeddedESName, &metav1.DeleteOptions{})
 			}
 		}()
-		esDeployment := NewESDeployment(namespace)
+		esDeployment := NewESDeployment(namespace, obj)
 		_, err = dep.Create(esDeployment)
-		if err != nil && !apierrors.IsAlreadyExists(err) {
-			return errors.Wrapf(err, "create deployment %s fail", loggingconfig.EmbeddedESName)
+		if err != nil {
+			if !apierrors.IsAlreadyExists(err) {
+				return errors.Wrapf(err, "create deployment %s fail", loggingconfig.EmbeddedESName)
+			}
+			if err = updateEmbeddedQuota(dep, obj); err != nil {
```

Comment:
Why we use update here?  The assumption of this block is that the deployment does not exist. So it not a valid case to update the deployment here

Reply:
The deployment is getting from cache, it may exist actually if the error does not exist, it will return before the update, so when it goes running the function updateEmbeddedQuota, the deployment exist.